woocommerce/grow
# Storybook Tool

### Changes proposed in this Pull Request:

Initial PR adding Project Templates as well as Storybook implementation

### Detailed test instructions:


1.  Uninstall all the storybook dependencies in GLA
2.  Install this branch in the project using `npm i -D  'https://gitpkg.now.sh/woocommerce/grow/storybook?add/storybook'`  
3. Check in `node_modules` you have `woocommerce-grow-story-book` folder
4.  Check in `node_modules/bin` you have `storybook.js` 
5. Replace storybook lines in scripts with this in GLA package JSON
```
"storybook": "storybook",
"storybook:build": "storybook build",
"storybook:deploy": "storybook deploy",
```
6. Execute `npm tun storybook` and see how the storybook watch initialises and you can access the URL
7. Execute `npm tun storybook:build` and see how the storybook generates dist directory
8. Execute `npm tun storybook:deploy` and see how the storybook deploys the dist version to GH pages in https://woocommerce.github.io/google-listings-and-ads/?path=/story/tree-select-control--base 
9. Check the templates generated with `yo:ventures`

### Additional details:

### Changelog entry

> Add - Storybook
> Add - Repository Templates
